Move methods to new API

Datseris opened the pull request:
I started moving some methods to the new API, feel free to move some as well in this PR until we got everything through.

kahaaga commented:
> I started moving some methods to the new API, feel free to move some as well in this PR until we got everything through.

Sure, I can do the remaining methods straight away.

Datseris commented:
The iaaft source is a bit too large, so I'll leave it to you.

Datseris commented:
I've added the method from the paper on evaluating a good parameter for `ρ`. I've also added an (extremely sloppy) note in the docs to add the new stuff there. 

I think after we get through this we should focus on #39 and get everything up and running.

I've recently tagged SignalDecomposition.jl. Although not directly related with this package, they are both packages targeting (nonlinear) timeseries analysis, and I thought it would be nice to announce them together on discourse. Once we feel that we are happy with this API and we have docs up and running, we can reflect on whether it is time to announce.

kahaaga commented:
> I've added the method from the paper on evaluating a good parameter for `ρ`. I've also added an (extremely sloppy) note in the docs to add the new stuff there.
> I think after we get through this we should focus on #39 and get everything up and running.
> 
> I've recently tagged SignalDecomposition.jl. Although not directly related with this package, they are both packages targeting (nonlinear) timeseries analysis, and I thought it would be nice to announce them together on discourse. Once we feel that we are happy with this API and we have docs up and running, we can reflect on whether it is time to announce.

Sounds like a good plan!

I'll tag you once I have the iaaft ready, which should be by the end of the day. I've also re-done tests with the new API and added a new block shuffle surrogate.

Datseris commented on src/methods/randomshuffle.jl:
```diff
@@ -7,9 +7,16 @@ This method destroys any linear
 """
 struct RandomShuffle <: Surrogate end
-surrogate(x, ::RandomShuffle) = randomshuffle(x)
+function surrogate(x, rs::RandomShuffle)
+    sg = surrogenerator(x, rs)
+    sg()
+end
```
@kahaaga I don't understand why you defined this method? It gives you literally nothing over the existing definition of `surrogate`.

You only had to delete the previous old method. If you think about how Julia's multiple dispatch works, it actually creates exactly the method you wrote based on the existing `surrogate` definition in the file `api.jl`.

kahaaga replied:
You are absolutely right, this is unneccesary. Removed.

kahaaga commented:
@Datseris The IAAFT method now works with the new api. I'm merging this, so we can start working on the documentation.
